osu.Game/Beatmaps/ControlPoints/ControlPointInfo.cs

```diff
@@ -12,6 +12,11 @@ namespace osu.Game.Beatmaps.ControlPoints
     [Serializable]
     public class ControlPointInfo
     {
+        private readonly TimingControlPoint timingSearchPoint = new TimingControlPoint();
```

Comment:
makes these methods non-threadsafe. probably worth adding to xmldoc or reconsidering.